code/modules/tgui/tgui_window.dm

```diff
@@ -246,24 +246,19 @@
  *
  * Makes an asset available to use in tgui.
  *
- * required asset
+ * required asset datum/asset
  *
  * return bool - TRUE if any assets had to be sent to the client
  */
-/datum/tgui_window/proc/send_asset(asset)
+/datum/tgui_window/proc/send_asset(datum/asset/asset)
 	if(!client || !asset)
 		return
-	if (istype(asset, /datum/asset))
-		var/datum/asset/current_asset = asset
-		sent_assets |= list(current_asset)
-		. = current_asset.send(client)
-		if(istype(current_asset, /datum/asset/spritesheet))
-			var/datum/asset/spritesheet/spritesheet = current_asset
-			send_message("asset/stylesheet", spritesheet.css_filename())
-		send_message("asset/mappings", current_asset.get_url_mappings())
-	else
-		client << browse_rsc(asset)
-		. = TRUE
+	sent_assets |= list(asset)
```

Comment:
I didn't think about this, because the asset datum is generated, they would tend to collect in this list as it builds overtime. The client datum has similar code

The solution would be to build a helper proc that either returns an already existing changelog_item datum or creates a new one.

https://github.com/tgstation/tgstation/blob/4e2a36fe7e56fad93495adf01c697f4b5de04e90/code/modules/asset_cache/asset_list.dm#L9-L16

Would be a good example